optimum/onnxruntime/modeling_decoder.py

```
@@ -625,15 +700,53 @@ def _from_pretrained(
+            decoder_with_past_path = None
+            if use_merged is False:
+                logger.info(
+                    "`use_merged` is automatically set to `True` as the decoder takes `use_cache` as an input."
+                )
+                use_merged = True
+        elif use_merged is True and decoder_with_past_path is not None:
+            try:
+                logger.info("Merging decoders...")
+                decoder_merged_path = decoder_path.parent / ONNX_DECODER_MERGED_NAME
+                merge_decoders(
+                    decoder=decoder_path,
+                    decoder_with_past=decoder_with_past_path,
+                    save_path=decoder_merged_path,
+                )
+                decoder_path = decoder_merged_path
+                decoder_with_past_path = None
+            except Exception as e:
+                logger.error(
+                    f"{e}.\nUnable to merge decoders. Will load two decoders for inference which will double the memory usage."
+                )
+                use_merged = False
+        elif use_merged is True and decoder_with_past_path is None:
+            raise FileNotFoundError(
+                f"The parameter `use_cache={use_cache}` and `use_merged=True` were passed to `ORTModelForConditionalGeneration.from_pretrained()`"
+                f" but the given decoder doesn't contain computation graphs for both w/o and w/. past scenarios."
+                f" And no ONNX file using past key values could be found in {str(Path(model_id, subfolder))} for merging."
             )
+        elif use_cache is True and decoder_with_past_path is None:
+            raise FileNotFoundError(
+                f"The parameter `use_cache=True` and `use_merged={use_merged}` were passed to ORTModelDecoder.from_pretrained()"
+                " but no ONNX file using only past key values could be found in"
+                f" {str(Path(model_id, subfolder))}, with the error:\n    {cache_error_log}"
```

Isn't it better to raise the errors ahead?